#! /usr/bin/env python

PACKAGE='tuw_image_publisher'
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("show_image",      bool_t,   0, "on true a opencv window will show the image", True )
gen.add("publish_rate",    double_t, 0, "publishing rate in Hz", 5, 0.1, 100)
gen.add("frame_id",        str_t,    0, "Frame to use for camera image", "camera")
gen.add("camera_info_url", str_t,    0, "Path to camera_info", "")
gen.add("folder",          str_t,    0, "folder with images (enviroment variables such as ${HOME} can be used",  "${HOME}/tmp")
gen.add("file_regx",       str_t,    0, "regex file expression (.*)bmp, (.*)png, (.*)jpg, (.*)$",  "(.*)png")
gen.add("camera_info",     str_t,    0, "camera_info",  "camera_info.yml")



exit(gen.generate(PACKAGE, "tuw_image_publisher", "ImagePublisher"))
